src/org/labkey/test/components/glassLibrary/grids/HeatMap.java

```diff
@@ -0,0 +1,211 @@
+package org.labkey.test.components.glassLibrary.grids;
+
+import org.labkey.test.Locator;
+import org.labkey.test.WebDriverWrapper;
+import org.labkey.test.components.Component;
+import org.labkey.test.components.WebDriverComponent;
+import org.openqa.selenium.By;
+import org.openqa.selenium.StaleElementReferenceException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.labkey.test.WebDriverWrapper.WAIT_FOR_JAVASCRIPT;
+import static org.labkey.test.util.TestLogger.log;
+
+public class HeatMap extends WebDriverComponent<HeatMap.ElementCache>
+{
+    private final WebElement _el;
+    private final WebDriver _driver;
+    private final Locator.XPathLocator _rowsLoc = Locator.tag("tbody").childTag("tr").withoutClass("grid-empty").withoutClass("grid-loading");
+    private final Locator _loadingRowLoc = Locator.css("tbody tr.grid-loading");
+    private final Locator _spinnerLoc = Locator.css("span i.fa-spinner");
+    private final Locator _emptyRowLoc = Locator.css("tbody tr.grid-empty");
+
+    protected HeatMap(WebElement element, WebDriver driver)
+    {
+        _el = element;
+        _driver = driver;
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return _el;
+    }
+
+    @Override
+    public WebDriver getDriver()
+    {
+        return _driver;
+    }
+
+    public String currentMonth()
+    {
+        return getColumnNames().get(12);
+    }
```

Comment:
We shouldn't assume that a HeatMap has 12 months, or that all HeatMaps always have months on the X-axis, really.

Reply:
OK, I'll take this out, leave the` getColumnByIndex()` method there, and refactor usages of `currentMonth()` to use it instead